index.js

```diff
@@ -38,16 +39,66 @@ module.exports.pitch = function(request) {
 			compilation.cache = compilation.cache[subCache];
 		}
 	});
+
+	// If the user doesn't include the magic `__assets__` anywhere, there is no
+	// need to produce an asset file.
+	var includeAssets = false
+
+	// I don't know how to register this in webpack to get a hashed filename.
+	// This is my temporary hack to use while I try to find a good solution.
+	var BAD_HASH = Math.random().toString(36).substring(2, 10)
+	// Service workers won't update if the file hasn't changed, so it's
+	// important that __assets__ changes any time the list of assets has
+	// changed. Ideally it would not change if the list of assets hasn't
+	// changed.
+	var assetListFilename = "assets." + BAD_HASH + ".js"
+
+	workerCompiler.parser.plugin("expression __assets__" , function(expr) {
+		includeAssets = true;
+		// importScripts can only be used inside a worker and it's much like a `<script>` tag.
+		const dep = new ConstDependency("(importScripts("  + JSON.stringify(assetListFilename) + "), __asset_list)", expr.range);
+		dep.loc = expr.loc;
+		this.state.current.addDependency(dep);
+		return true
+	})
+
+	// I know this is hacky, but I don't know how else to do this. In my (weak)
+	// defense... it is documented
+	// http://webpack.github.io/docs/loaders.html#_compiler
+	this._compiler.plugin('emit', function(compilation, callback) {

```

Comment:
`this._compiler` is hacky according to the documentation. Is there another way I could get the complete list of files from the parent compiler? Is there a better way I can add this new asset that contains the list of files?